src/devices/cpu/tms34010/34010ops.hxx

```diff
@@ -2395,7 +2399,40 @@ void tms340x0_device::rpix_b(uint16_t op)
 void tms340x0_device::setcdp(uint16_t op)
 {
 	if (!m_is_34020) { unimpl(op); return; }
-	logerror("020:setcdp\n");
+	off_t dptch = DPTCH();
+
+	// logerror("020:setcdp: PC=%x, DPTCH()=%x\n", m_pc, dptch);
+
+	// Check whether we're dealing with an even number
+	if ((dptch & 1) == 0) {
+		switch(population_count_32(dptch))
+		{
+			// .. only single bit set, pitch is power of two!
+			case 1:
+			{
+				uint32_t val = log2(dptch);
```

Comment:
You can get this from counting leading zeroes.  You don't want to invoke floating-point maths.